scala-thoughts.md

```diff
@@ -0,0 +1,13 @@
+## Initial Scalal Thoughts
+
+In some ways I really liked Scala. It felt very clean and intuitive, and the ability to
+to pattern match in a more functional way made a lot of sense while I was thinking about
+how to solve problems. That being said, since there wasn't always a lot of rigid syntax,
+I often had got confused between this and the language R, since I'm also learning that at
+the same time. There are a few oddities about R, Scala, and Python that I kept mixing up. 
+Variable assignment was a big one that distinguishes all three, but would sometimes catch
+myself doing the wrong one. Same with functions and parameters. I really liked Scala's way
+of defining functions and how you included the types. But since that isn't common in most
+of the other languages that I use on a daily basis, it was easy to forget that I had to.
+I could see why a designer might choose to do it that way, and it did help a lot with
+debugging type errors. 
```

Comment:
I agree; I also found myself typing Haskell and/or Python code since those are the languages I'm using the most right now. It's difficult because Scala seems similar to them in so many ways, but it only takes one wrong keyword to throw everything off. I also agree that the pattern match feature is a big plus! It might not make things concise, but it keeps things readable which is arguably more important.